connector/protobuf/src/test/scala/org/apache/spark/sql/protobuf/ProtobufFunctionsSuite.scala

```diff
@@ -1124,7 +1124,78 @@ class ProtobufFunctionsSuite extends QueryTest with SharedSparkSession with Prot
     }
   }
 
-  test("Corner case: empty recursive proto fields should be dropped") {
+  test("Retain empty proto fields when retain.empty.message.types=true") {
+    // When retain.empty.message.types=true, empty proto like 'message A {}' can be retained as
+    // a field by inserting a dummy column as sub column.
+    val options = Map("recursive.fields.max.depth" -> "4", "retain.empty.message.types" -> "true")
```

Comment:
Remove `"recursive.fields.max.depth" -> "4"` ? This test does not use recursive protobufs.

Reply:
There is a EmptyProtoWrapper below which is nested.